src/supervised/other.jl

```diff
@@ -26,12 +27,21 @@ isconvex(::PoissonLoss) = true
 isstronglyconvex(::PoissonLoss) = false
 
 # ===============================================================
-# L(target, output) = - target*ln(output) - (1-target)*ln(1-output)
+
+doc"""
+    CrossentropyLoss <: SupervisedLoss
+
+Cross-entropy loss also known as log loss and logistic loss is defined as:
+
+``L(target, output) = - target*ln(output) - (1-target)*ln(1-output)``
+"""
 
 immutable CrossentropyLoss <: SupervisedLoss end
 typealias LogitProbLoss CrossentropyLoss
 
 function value(loss::CrossentropyLoss, target::Number, output::Number)
+    target >= 0 && target <=1 || error("target must be in [0,1]")
+    output >= 0 && output <=1 || error("output must be in [0,1]")
```

Comment:
@Evizero @tbreloff 
Agreed with your concerns, but I think checks are necessary (espescially for libraries which other libraries are going to use). I would prefer an error rather than a strange behaviour in my code.

There are checks in `PeriodicLoss`, `HuberLoss` and epsilon losses.

Though I am not much experienced but in [SymPy](github.com/sympy/sympy) they have used checks for every possible corner case. Also Julia is awesome in showing errors:
```
ERROR: output must be in [0,1]
in value(::LossFunctions.CrossentropyLoss, ::Int64, ::Int64) at /home/nishnik/.julia/v0.5/LossFunctions/src/supervised/other.jl:44
```
Sorry, I am not much experienced in Julia to strongly pitch this, it is just my thinking. If you want me to remove checks I would do it, but I am a little reluctant to do this.
Otherwise ready to merge from my side.